### PR TITLE
Protect dependent jobs from removal

### PR DESCRIPTION
**Pull request by nulltoken**
## Pull request description

Another corner case discovered while working on #255: We can't blindly remove a `JobDefinition` when it's used for a dependent job.

Also fix the intent of a test that doesn't seem to be properly expressed. A dependent job cannot have a schedule. As such, it shouldn't be retrievable through `IRuntimeRegistry.TryGetSchedule()`.

### PR meta checklist
- [x] Pull request is targeted at `main` branch for code   
- [x] Pull request is linked to all related issues, if any.

### Code PR specific checklist
- [x] My code follows the code style of this project and AspNetCore coding guidelines.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] I have updated the appropriate sub section in the _CHANGELOG.md_.
- [x] I have added, updated or removed tests to according to my changes.
  - [x] All tests passed.

### CODE REVIEW

File: src/NCronJob/Registry/JobRegistry.cs [hunk 1]
```diff
@@ -45,19 +48,44 @@ public int GetJobTypeConcurrencyLimit(string jobTypeName)
...
         foreach (var jobDefinition in jobDefinitions)
         {
             Remove(jobDefinition);
         }
+
+        return jobDefinitionFullName;
     }
 
-    public JobDefinition AddDynamicJob(
+    public static JobDefinition CreateDynamicJob(
```
Comment (linkdotnet) [inline src/NCronJob/Registry/JobRegistry.cs:88]: That is information which is solely relying on `JobDefinition` - wouldn't it make more sense to move it there?

File: src/NCronJob/Registry/JobRegistry.cs [hunk 2]
```diff
@@ -108,6 +134,18 @@ private JobDefinition[] FilterByAndProject(
         ? []
         : transform(types).ToArray();
 
+    private void EnsureCanBeRemoved(Func<JobDefinition, bool> jobDefintionFinder)
```
Comment (linkdotnet) [inline src/NCronJob/Registry/JobRegistry.cs:137]: Very good point

File: src/NCronJob/Registry/JobRegistry.cs (hunk 1 above)
Comment (nulltoken) [reply src/NCronJob/Registry/JobRegistry.cs]: Fixed. It was actually simpler to just inline it.

**Review by linkdotnet — APPROVED** [review]
As alawys, love it - much appreaciated
